polygeist/lib/polygeist/Ops.cpp

```diff
@@ -1173,7 +1173,9 @@ class Memref2PointerIndex final : public OpRewritePattern<Memref2PointerOp> {
       return failure();
 
     auto MET = src.getSource().getType().cast<MemRefType>().getElementType();
-    if (MET.isa<LLVM::LLVMStructType>())
+    // SYCL types are lowered to LLVM struct type.
+    bool isSYCLTy = MET.getDialect().getNamespace().contains("sycl");
```

Comment:
We have `sycl::isSYCLType` in the `sycl` dialect (`mlir/Dialect/SYCL/IR/SYCLOpsTypes.h`)

Reply:
Unfortunately, I cannot use  `sycl::isSYCLType`, doesn't link. Maybe it will work after your restructuring?

Reply:
True, it might. I will take a look when the PR is merged. Thanks